spec/rester/client_spec.rb

```diff
@@ -256,243 +273,242 @@ def let_retry_period_pass
         after { let_retry_period_pass; success_request }
 
         it 'should log that circuit is now closed' do
-          expect(logger).to receive(:info).with('circuit closed').once
+          correlation_id = Rester.correlation_id
+          producer = Rester.request_info[:producer_name]
+          consumer = Rester.request_info[:consumer_name]
+
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{correlation_id} Consumer=#{consumer} " \
+            "Producer=#{producer} GET /v1/ping - circuit closed for " \
+            "DummyService"
+          ).once
         end
       end # with error_threshold reached
-    end # #request
 
-    describe '#tests', :tests do
-      let(:tests) { client.tests(*args) }
-      subject { tests }
+      context 'before first real request' do
+        before { success_request }
 
-      context 'without connection' do
-        context 'with string argument' do
-          let(:args) { ['token'] }
+        it 'should default producer name to to the service after first ping' do
+          expect(client.send(:_producer_name)).to eq "DummyService"
+        end
+      end # without producer name set
 
-          describe '#get' do
-            subject { tests.get }
-            it { expect { subject }.to raise_error RuntimeError, 'not connected' }
-          end
+      context 'with producer name set' do
+        before { success_request }
 
-          describe '#update' do
-            subject { tests.update }
-            it { expect { subject }.to raise_error RuntimeError, 'not connected' }
-          end
+        it 'should default producer name to "Producer"' do
+          expect(client.send(:_producer_name)).to eq "DummyService"
+        end
```

Comment:
Should this test be renamed?  It's no longer defaulting to producer.  Also this is a private method being accessed here.

Maybe it should just be `#name`?  So you could do `MyServiceClient.name => "MyService"`.